docs/contributing/coding-guidelines/ros-nodes/topic-message-handling/index.md

```diff
@@ -0,0 +1,200 @@
+The manner will be effective when there is a large difference between reception frequency and consumption frequency. For example, even if a message, such as a CAN message, is received at higher than 100 Hz, a user logic consumes messages at slower frequency such as 10 Hz. In such a case, the user logic should retrieve the required number of messages with the `take()` method to avoid the unexpected overhead.
+
+### 3. Obtain data by calling `Subscription->take` and then call a callback function
+
+You can combine the `take()` (strictly `take_type_erased()`) method and the callback function to process received messages in a consistent way. Using this combination does not require waking up a thread.
+Here is a sample code snippet from [_ros2_subscription_examples/simple_examples/src/timer_listener_using_callback.cpp_](https://github.com/takam5f2/ros2_subscription_examples/blob/main/simple_examples/src/timer_listener_using_callback.cpp).
+
+```c++
+      auto msg = sub_->create_message();
+      rclcpp::MessageInfo msg_info;
+      if (sub_->take_type_erased(msg.get(), msg_info)) {
+        sub_->handle_message(msg, msg_info);
+
+```
+
+In the code above, a message is taken by the `take_type_erased()` method before a registered callback function is called via the `handle_message()` method. Note that you must use `take_type_erased()` instead of `take()`. `take_type_erased()` needs `void` type data as its first argument. You must use the `get()` method to convert `msg` whose type is `shared_ptr<void>` to `void` type. Then the `handle_message()` method is called with the obtained message. A registered callback function is called within `handle_message()`.
+You don't need to take care of message type which is passed to `take_type_erased()` and `handle_message()`. You can define the message variable as `auto msg = sub_->create_message();`.
+You can also refer to [_the API document_](http://docs.ros.org/en/humble/p/rclcpp/generated/classrclcpp_1_1SubscriptionBase.html#_CPPv4N6rclcpp16SubscriptionBase16take_type_erasedEPvRN6rclcpp11MessageInfoE) as for `create_message()`, `take_type_erased()` and `handle_message()`.
+
+### 4. Obtain data by a callback function
+
+A conventional manner, typically used in ROS 2 application, is a message reference using a callback function, is available. If you don't use a callback group with `automatically_add_to_executor_with_node = false`, a registered callback function will be called automatically by an executor when a topic message is received.
+One of the advantages of this manner is that you don't have to take care whether a topic message is passed through inter-process or intra-process. Remember that `take()` can only be used for inter-process communication via DDS, while another manner provided by `rclcpp` can be used for intra-process communication via `rclcpp`.
+
+## Appendix
```

Comment:
May be you could also add the links to example PRs that migrates from conventional method to recommended method under appendix. For example,  https://github.com/autowarefoundation/autoware.universe/pull/6702/files

Reply:
@mitsudome-r 
Thank you very much.
I took your point in https://github.com/autowarefoundation/autoware-documentation/pull/543/commits/7b98baa3ac28e8af2746395dcd1f7fb14efc98dd.
I was going to put the sentence "The recommended manner is already in use in the autoware.universe" first, but decided against it because it would be inappropriate considering that this document will be read after many applications have been made.

Reply:
@mitsudome-r 

I committed index.md again in https://github.com/autowarefoundation/autoware-documentation/pull/543/commits/a7bf42d674869d70940a17fd93feb74358abcedc.